### DISCUSSION TITLE
Thread data manual store

### DISCUSSION THREAD

**Discussion by hani-n** [openthread/openthread#9336] [Q&A] [answered]
Hello Everyone,

## Context
Our platform is ESP32 Host Processor + EFR32 RCP.
We are trying to manually safe thread dataset to flash the first time a device successfully joins a network via joiner,
or when a device successfully forms a network.

## What we need help with
During testing we observe our `otStateChangedCallback`  receive (amongst others) a `0x200` flag (i.e aFlags == `OT_CHANGED_THREAD_NETDATA)
which we have been experimenting with as a trigger to store Thread data.

How safe is it to assume that we would only get this flag once a network is successfully formed (attached to) and that the dataset is ready to be saved?
Any insights on what other network event might trigger this flag to be received alone?

**Comment by abtink** [comment, marked as answer]
The OpenThread stack already manages this and saves all information that needs to be persisted, including the Operational Dataset, in non-volatile settings using the platform [`otPlatSettings APIs`](https://github.com/openthread/openthread/blob/992be2781f938b5208489dae79575b5e4f08e2a1/src/core/common/settings.hpp). This includes initial Dataset configuration and any changes made later to it or other persisted entries.

Also:
- `OT_CHANGED_ACTIVE_DATASET` and `OT_CHANGED_PENDING_DATASET`  flags signal the Active or Pending Operational Dataset change.
- `OT_CHANGED_THREAD_NETDATA` flag is related to the Thread Network Data, which contains information such as on-mesh prefixes, off-mesh routes, and services. *The Network Data should not be saved in settings*. It is maintained by Border Routers (BRs) and the Leader and distributed to all nodes.

**Reply by hani-n** [reply]
Understood, thanks for the explanation
